spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -613,8 +611,11 @@ def update_availability!(app_id, availability)
       # API will error out if cleared_for_preorder is false and app_available_date has a date
       cleared_for_preorder = availability.cleared_for_preorder
       app_available_date = cleared_for_preorder ? availability.app_available_date : nil
+      data["b2bAppEnabled"] = availability.b2b_app_enabled
+      data["educationalDiscount"] = availability.educational_discount
       data["preOrder"]["clearedForPreOrder"] = { "value" => cleared_for_preorder, "isEditable" => true, "isRequired" => true, "errorKeys" => nil }
       data["preOrder"]["appAvailableDate"] = { "value" => app_available_date, "isEditable" => true, "isRequired" => true, "errorKeys" => nil }
+      data["b2bUsers"] = availability.b2b_app_enabled ? availability.b2b_users.map { |user| { "value" => { "add" => user.add, "delete" => user.delete, "dsUsername" => user.ds_username } } } : []
```

Comment:
I believe you can convert the `b2b_users` back into data by doing: `b2b_users.raw_data.to_h`. Ideally, the model and the `attr_mapping` should be the only place that has to know how to convert to and from the JSON data.

Reply:
Okay digging into the metaprogramming ya'll have to make sense of it. I tried to use the mapping but it was not working quite as expected - it might be the limitation of the model. 

I am more than happy to give it a proper go, however. If it is limited in scope, I can make the changes here but if it is extensive, hopefully be I can make it in another PR.

Reply:
Hey @heartofagoof, if it doesn't work as expected out of the box, I wouldn't bother too much trying to hack it to work

Reply:
Yeah, no worries at all! We will merge what you got here 😊